src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,119 @@
+        int threshold = (int) (table.length * LOAD_FACTOR);
+        if (currentNode == null) {
+            table[i] = newNode;
+            if (++size > threshold) {
+                resize();
+            }
+            return;
+        }
+        do {
+            if (keyCheck(key, currentNode)) {
+                currentNode.value = value;
+                return;
+            }
+            if (currentNode.next == null) {
+                currentNode.next = newNode;
+                if (++size > threshold) {
+                    resize();
+                }
+                return;
+            }
+            currentNode = currentNode.next;
+        } while (true);
     }
 
     @Override
     public V getValue(K key) {
+        if (table != null) {
+            for (Node<K, V> currentNode : table) {
+                if (currentNode != null) {
+                    do {
+                        if (keyCheck(key, currentNode)) {
+                            return currentNode.value;
+                        }
+                        currentNode = currentNode.next;
+                    } while (currentNode != null);
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private int hash(Object key) {
+        int hash;
+        return (key == null) ? 0 : (hash = key.hashCode()) ^ (hash >>> 16);
+    }
+
+    private int index(int hash) {
+        return hash & (table.length - 1);
+    }
+
+    private void resize() {
+        if (table.length < MAX_CAPACITY) {
+            Node<K, V>[] oldTable = table;
+            table = new Node[table.length << 1];
+            for (Node<K, V> oldNode : oldTable) {
+                while (oldNode != null) {
+                    Node<K, V> nodeToPut = new Node<>(
+                            oldNode.hash, oldNode.key, oldNode.value, null);
+                    int newIndex = index(oldNode.hash);
+                    Node<K, V> newNode = table[newIndex];
+                    if (newNode == null) {
+                        table[newIndex] = nodeToPut;
+                    } else {
+                        do {
+                            if (newNode.next == null) {
+                                newNode.next = nodeToPut;
+                                break;
+                            }
+                            newNode = newNode.next;
+                        } while (true);
+                    }
+                    oldNode = oldNode.next;
+                }
+            }
```

Comment:
could we just go through all nodes in out table and call put(K key, V value) method?

Reply:
Yes, we can, but in this case we will do redundant checks for the key equality and resize. The key equality was already done when the old table was filling. Also there's no point in checking the size because we are currently increasing the table in a twice after placing just only one element.

Reply:
By then you have duplicated logic in `put(K key, V value)` method and here, lets move it in separate method than

Reply:
Done